toolchain/check/eval.cpp

```diff
@@ -1272,4 +1401,48 @@ auto TryEvalInst(Context& context, SemIR::InstId inst_id, SemIR::Inst inst)
   return SemIR::ConstantId::NotConstant;
 }
 
+auto TryEvalInst(Context& context, SemIR::InstId inst_id, SemIR::Inst inst)
+    -> SemIR::ConstantId {
+  EvalContext eval_context = {
+      .context = context,
+      .specific_id = SemIR::GenericInstanceId::Invalid,
+      .specific_eval_info = nullptr,
+  };
+  return TryEvalInstInContext(eval_context, inst_id, inst);
+}
+
+auto TryEvalBlockForSpecific(Context& context,
+                             SemIR::GenericInstanceId specific_id,
+                             SemIR::GenericInstIndex::Region region)
+    -> SemIR::InstBlockId {
+  auto generic_id = context.generic_instances().Get(specific_id).generic_id;
+  auto eval_block_id = context.generics().Get(generic_id).GetEvalBlock(region);
+  auto eval_block = context.inst_blocks().Get(eval_block_id);
+
+  llvm::SmallVector<SemIR::InstId> result;
+  result.resize(eval_block.size(), SemIR::InstId::Invalid);
+
+  SpecificEvalInfo eval_info = {
+      .region = region,
+      .values = result,
+  };
+  EvalContext eval_context = {
+      .context = context,
+      .specific_id = specific_id,
+      .specific_eval_info = &eval_info,
+  };
```

Comment:
FWIW, the extra struct from SpecificEvalInfo had made me expect it'd be replaced and/or separately used, but this is the only assignment I see other than `nullptr`. Given the way EvalContext is constructed, had you considered moving its members over, or perhaps making it `std::optional` instead of `*`?

Reply:
Originally I was concerned about this being large (containing a `SmallVector`) and I didn't want to include it in `EvalContext` for that reason, but that ended up not being a problem. Turned into a `std::optional`.